content/protobuf/language-mapping/service.md

```diff
@@ -0,0 +1,220 @@
+
+```csharp
+public readonly partial record struct WidgetProxy : IWidget
+{
+    public WidgetProxy(
+        IInvoker invoker,
+        ServiceAddress? serviceAddress = null,
+        ProtobufEncodeOptions? encodeOptions = null)
+    {
+        ...
+    }
+
+    public WidgetProxy(
+        IInvoker invoker,
+        System.Uri serviceAddressUri,
+        ProtobufEncodeOptions? encodeOptions = null)
+        : this(invoker, new ServiceAddress(serviceAddressUri), encodeOptions)
+    {
+    }
+}
+```
+
+The `invoker` parameter represents your [invocation pipeline](/icerpc/invocation/invocation-pipeline), the
+`serviceAddress` or `serviceAddressUri` parameter corresponds to the
+[address](/icerpc/invocation/service-address) of the remote service, and the `encodeOptions` parameter allows
+you to customize the encoding of your Protobuf messages. See [ProtobufEncodeOptions] for details.
+
+A `null` service address is equivalent to an icerpc service address with the default path of the associated Protobuf
+service.
+
+{% callout type="note" %}
+The default path of a Protobuf service is `/` followed by its fully qualified name. For example, the default path of
+Protobuf service `visitor_center.Greeter` is `/visitor_center.Greeter`.
+{% /callout %}
+
+### I*Name*Service
+
+Interface I*Name*Service is a server-side helper: it helps you create a service (a C# class) that implements Protobuf
+service _Name_.
+
+The principle is straightforward: your service class must be a partial class that implements I*Name*Service. It must
+also carry the [ProtobufService] attribute.
+
+The `ProtobufService` attribute instructs the Protobuf Service source generator to implement interface [IDispatcher] by
+directing incoming requests to I*Name*Service methods based on the Protobuf rpc method names.
+
+For example:
+
+{% side-by-side alignment="top" %}
+
+```protobuf
+syntax = "proto3";
+package example;
+option csharp_namespace = "Example";
+
+service Widget {
+    rpc Spin (SpinConfig)
+        returns (google.protobuf.Empty);
+}
+```
+
+```csharp
+namespace Example;
+
+// Generated code
+public partial interface IWidgetService
+{
+    // One method per operation
+    ValueTask<WellKnownTypes.Empty> SpinAsync(
+        SpinConfig message,
+        IFeatureCollection features,
+        CancellationToken cancellationToken);
+}
+
+// Application code
+[ProtobufService]
+internal partial class MyWidget : IWidgetService
+{
+    // implement SpinAsync ...
+}
+```
+
+{% /side-by-side %}
+
+{% callout type="note" %}
+Even though I*Name*Service is an interface, it's not used as an abstraction: you shouldn't make calls to this interface
+or create decorators for this interface. It's just a model that your service class must implement.
+{% /callout %}
+
+Note that the same service class can implement any number of Protobuf services provided their rpc methods have unique
```

Comment:
Some kind of separator here would help make it more readable.
Since it's a separate conditional clause.
Otherwise it read to me like the services are being provided by something.
```suggestion
Note that the same service class can implement any number of Protobuf services, provided their rpc methods have unique
```